gsy_framework/validators/profile_validator.py

```diff
@@ -0,0 +1,44 @@
+from collections import OrderedDict
+from datetime import timedelta
+from typing import Dict
+
+from pendulum import DateTime, duration
+
+from gsy_framework.utils import convert_str_to_pendulum_in_dict
+
+
+class ProfileValidator:
+    """Validate profiles"""
+
+    @classmethod
+    def validate(cls, profile: Dict[str, float], start_time: DateTime, end_time: DateTime,
+                 slot_length: timedelta):
+        """Validate if profile corresponds to the start_time, end_time and slot_length setting."""
+        assert len(profile) > 0, "profile is empty"
+        ordered_profile = OrderedDict(convert_str_to_pendulum_in_dict(profile))
+        cls._validate_start_end_date(ordered_profile, start_time, end_time, slot_length)
+        cls._validate_slot_length(ordered_profile, slot_length)
+
+    @staticmethod
+    def _validate_start_end_date(
+            profile: Dict[DateTime, float], start_time: DateTime, end_time: DateTime,
+            slot_length: timedelta):
+        profile_start_time = next(iter(profile))
+        profile_end_time = next(reversed(profile))
+        assert profile_start_time <= start_time, (
+            f"start_time is not valid {profile_start_time}, should be <= {start_time}")
+        if profile_end_time - profile_start_time == duration(days=1) - slot_length:
```

Comment:
This is an exact comparison though, and I do not recall the exact condition when we copy to other days. Do we also check out for exactly one day in that part of the code too ?

Reply:
> I do not recall the exact condition when we copy to other days

we do here, here https://github.com/gridsingularity/gsy-framework/blob/285da741929e5731ebeb3caf3dd52c75a47585b2/gsy_framework/read_user_profile.py#L331

> Do we also check out for exactly one day in that part of the code too 

No, we do not, This would break the gsy-e that still allows gaps in the profiles.

Reply:
If the end_time is not the expected end_time, the profile should be rejected.
Either it is exactly one day (minus the slot_length) or it is as long as the simulation duration.
If you like I could make it a bit less strict and also allow profiles that are longer than one day but shorter than the simulation duration. However, this will mess with the `copy_profile_to_multiple_days` here we only look at the hours and minutes and the latest available value that corresponds to the hour:minute combination will be used. So, if you just add one more time_slot to the profile (for the next days' 00:00), it will set the 00:00 to the next days, but not for the first day. IMO this is not ideal and I rather be strict here.
What do you think?

Reply:
Convinced, better to remain strict here indeed. Thanks for the detailed explanation!